src/v/hashing/secure.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include "bytes/bytes.h"
+
+#include <gnutls/crypto.h>
+#include <gnutls/gnutls.h>
+
+class hmac_exception : public std::exception {
+public:
+    explicit hmac_exception(const char* msg)
+      : _msg(msg) {}
+
+    const char* what() const noexcept final { return _msg; }
+
+private:
+    const char* _msg;
+};
+
+namespace internal {
+
+template<gnutls_mac_algorithm_t Algo, size_t DigestSize>
```

Comment:
Did you consider using nettle directly, or crypto++?

Reply:
no i just went with the first dependency we had that'd work. i did try to make our interface agnostic so we swap out the implementation in the `internal` namespace. so definitely happy to replace later.